core/blockchain.go

```diff
@@ -963,12 +964,21 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (int, error) {
 			events = append(events, ChainSplitEvent{block, logs})
 		}
 		stats.processed++
+
+		// Update the current gas limit (mining only) based on import times
+		params.CurrentGasCeilLock.Lock()
+		if time.Since(bstart) > params.BlockTimeLimit {
+			params.CurrentGasCeil = new(big.Int).Div(params.CurrentGasCeil, params.CurrentGasCeilDiv)
```

Comment:
I was also thinking that we could just point to to 0 since it would take a lot of time for the chain to react anyway.